components/org.wso2.carbon.appmgt.gateway/src/main/java/org/wso2/carbon/appmgt/gateway/handlers/security/saml2/SAML2AuthenticationHandler.java

```diff
@@ -271,6 +277,26 @@ public boolean handleRequest(MessageContext messageContext) {
         }
     }
 
+    public static TokenGenerator getTokenGenerator() {
```

Comment:
1. Need to change private void initTokenGenerator, as this is private use, and not static.
2. Improve JWT generator wiring with proper OSGI scr.ref assignment.